python/pyspark/sql/connect/dataframe.py

```diff
@@ -511,7 +511,7 @@ def drop(self, *cols: "ColumnOrName") -> ParentDataFrame:  # type: ignore[misc]
             session=self._session,
         )
 
-    def filter(self, condition: Union[Column, str]) -> ParentDataFrame:
```

Comment:
`str` type here means a SQL expression, not a column name

Reply:
Is that mean the behavior is different Classic vs Connect?

Reply:
no, I think we should change it in all 3 `dataframe.py` to `def filter(self, condition: Union[Column, str])`.

Reply:
I think he meant to keep the type hints because you're changing it to `ColumnOrName` which means a column name. However, the actual argument it takes is a SQL expression string.

Reply:
Got it. Thanks for the clarification!